### PR TITLE
Feature/print output

### PR DESCRIPTION
**Pull request by zorafuchs**

### CODE REVIEW

File: lib/renuo/bin-check/master_thread.rb [hunk 1]
```diff
@@ -2,11 +2,12 @@
 
 module RenuoBinCheck
   class MasterThread
-    attr_reader :threads
+    attr_reader :threads, :printer
```
Comment (lukaselmer) [inline lib/renuo/bin-check/master_thread.rb]: do you really want to call the class ```OutputPrinter``` if you call the variable ```printer```? ;)

File: lib/renuo/bin-check/output_printer.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module RenuoBinCheck
+  class OutputPrinter
+    def print_output(results)
+      print_result(results, 0) { |result| $stdout.puts result.output }
+    end
+
+    def print_error_output(results)
```
Comment (lukaselmer) [inline lib/renuo/bin-check/output_printer.rb]: wouldn't this only print the failing result?

File: lib/renuo/bin-check/master_thread.rb [hunk 3]
```diff
@@ -21,8 +22,20 @@ def finalize
       until waiter.empty?
         thread = waiter.next_wait
         @results << thread[:result]
-        exit 1 if @results.last.exit_code == 1
+        exit_with_error if @results.last.exit_code == 1
```
Comment (lukaselmer) [inline lib/renuo/bin-check/master_thread.rb]: don't we also want to give developer feedback asap? then we may also want to print if a check ran successfully?

File: spec/renuo/bin-check/initializer_spec.rb [hunk 4]
```diff
@@ -29,22 +29,22 @@
       end
 
       begin
-        bin_check.run
+        expect { bin_check.run }.to output("I have been running\nNow I'm tired\n").to_stdout
```
Comment (lukaselmer) [inline spec/renuo/bin-check/initializer_spec.rb]: is this an integration test or a unit test? i guess it is a unit test, and therefore i wouldn't test if this is the output, but instead if the printer was called with the correct object.

File: spec/renuo/bin-check/output_printer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
+require './lib/renuo/bin-check/output_printer'
+
+RSpec.describe RenuoBinCheck::OutputPrinter do
```
Comment (lukaselmer) [inline spec/renuo/bin-check/output_printer_spec.rb]: very good test 👍

File: lib/renuo/bin-check/output_printer.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      print_result(results, 1) { |result| $stderr.puts result.error_output }
+    end
+
+    private
+
+    # :reek:ControlParameter
+    # :reek:UtilityFunction
+    def print_result(results, error_code)
+      results.each do |result|
+        yield(result) if result.exit_code == error_code
```
Comment (lukaselmer) [inline lib/renuo/bin-check/output_printer.rb]: this looks like a hack ```result.exit_code == error_code```

Comment (lukaselmer) [inline lib/renuo/bin-check/output_printer.rb]: btw: 2,3,234 could also be error codes - you must never check if the error code = 1 if you want to detect an error

**Comment by lukaselmer** [comment]
still needs some work

**Comment by lukaselmer** [comment]
but most of it lgtm :)

File: spec/renuo/bin-check/initializer_spec.rb (hunk 4 above)
Comment (zorafuchs) [inline spec/renuo/bin-check/initializer_spec.rb]: They are integration tests

File: lib/renuo/bin-check/master_thread.rb (hunk 3 above)
Comment (zorafuchs) [inline lib/renuo/bin-check/master_thread.rb]: I thought you only wanted it to show the failed one? and I also that i should wait till the end and print all of them then? I thought we said it like that when we were talking about the ipa task, but I can also change it :)

File: lib/renuo/bin-check/master_thread.rb (hunk 1 above)
Comment (zorafuchs) [inline lib/renuo/bin-check/master_thread.rb]: I renamed the class

**Pull request merged**